Fix enterFrame clearing timers too soon

XeduR opened the pull request:
Since the enterFrame function is the first to be run, it kept clearing the timer which had just fired, which made it impossible to control this specific timer from within its own listener.

Now the enterFrame function will fire a timer's listener before removing the timer from runlist, which allows the user to control the timer from its listener.


_table.insert and table.remove functions were also localised to improve their performance slightly._

This closes https://github.com/coronalabs/corona/issues/162

XeduR committed:
Fix enterFrame clearing timers too soon

Since the enterFrame function is the first to be run, it kept clearing the timer which had just fired, which made it impossible to control this specific timer from within its own listener.

Now the enterFrame function will fire a timer's listener before removing the timer from runlist, which allows the user to control the timer from its listener.

-
table.insert and table.remove functions were also localised to improve their performance slightly.

lifestreamer commented:
Thank you for the fix, but I'm sorry to tell you that it only fixes half of the problem.
It fixes pausing the current timer from its callback. Pausing previous timers, all timers or all timers by tag still wont work properly.

I think that enterFrames toInsert table would have to be visible to pause/resume functions so it can loop over it like it does over runlist to make things work.

Can you please take another look at this? Thanks.

```
my_timer = timer.performWithDelay(0, function()
	print("Timer 1 running")
end, -1)

timer.performWithDelay(0, function()
	print("Pausing timer 1")
	timer.pause(my_timer)
end, 1)
```

XeduR committed:
Fix remaining known issues with timer functions

Pausing timers using their explicit handles is now handled correctly.

Using tags or cancelAll/pauseAll also now correctly take into account any timers that may have been created on the same frame as the cancel/pause command is given, but haven't yet been added to the Runlist. This ensures that all timers will be paused/cancelled as expected.
